analysis/priestholy/src/modules/shadowlands/items/FlashConcentration.tsx

```diff
@@ -0,0 +1,228 @@
+   * first Flash Heal generates a ApplyBuffStack or ApplyBuff event
+   * Note 2: Technically, the only time this function does something is on the first
+   * apply buff if there was no prepull FC.
+   * @param event with info about first stack, or prepull incomplete indication
+   */
+  onByPlayerFlashConcentrationInitialBuff(event: ApplyBuffEvent) {
+    if (event.prepull !== true) {
+      this._isFullStack = false;
+      this._currentStacks = 1;
+    }
+  }
+
+  /**
+   * Increments _totalDrops for our suggestion, since that should ideally not happen
+   */
+  onByPlayerFlashConcentrationBuffRemoval(event: RemoveBuffEvent) {
+    this._totalDrops += 1;
+    this._isFullStack = false;
+    this._currentStacks = 0;
+  }
+
+  /**
+   * when at full stack, it refreshes the last refresh timer,
+   * and counts a Wasteful Cast if it was used earlier than needed
+   */
+  onByPlayerFlashHealCast(event: CastEvent) {
+    if (this._isFullStack) {
+      if (event.timestamp - this._lastFullStackRefresh < 15000) {
+        this._wastefulCasts += 1;
+      }
+      this._lastFullStackRefresh = event.timestamp;
+    }
+  }
+
+  onByPlayerHealAmount(event: HealEvent) {
+    this._totalHeal += event.amount;
+    this._gainedHeal +=
+      (event.amount * (this._currentStacks * 0.03)) / (1 + this._currentStacks * 0.03);
+    this._gainedCastTime += this._currentStacks * 200;
+  }
+
+  onByPlayerHealCast(event: CastEvent) {
+    this._totalCastTime +=
+      event.timestamp -
+      (event.channel?.beginChannel?.timestamp
+        ? event.channel.beginChannel.timestamp
+        : event.timestamp);
+  }
+
+  statistic() {
+    // HealHPSNoFC gives the HPS which would have been obtained by casting the same number of heals without Flash Concentration Buff
+    const HealHPSNoFC =
+      (this._totalHeal - this._gainedHeal) / (this._totalCastTime + this._gainedCastTime);
+    // BuffedHealHPS gives the total HPS of Heal in this fight
+    const BuffedHealHPS = this._totalHeal / this._totalCastTime;
+    return (
+      <Statistic
+        size="flexible"
+        category={STATISTIC_CATEGORY.ITEMS}
+        tooltip={
+          <>
+            Impact is approximative since <SpellLink id={SPELLS.FLASH_CONCENTRATION.id} /> modifies
+            the gameplay in depth
+          </>
+        }
+      >
+        <BoringSpellValueText spell={SPELLS.FLASH_CONCENTRATION}>
+          <ItemHealingDone
+            // approximate considering the player would have cast heal without FC
+            amount={(BuffedHealHPS - HealHPSNoFC) * this._totalCastTime}
+            approximate
+          />
+          <br />
+          {this._gainedCastTime / 1000} sec saved on <SpellLink id={SPELLS.GREATER_HEAL.id} /> casts
+        </BoringSpellValueText>
+      </Statistic>
+    );
+  }
+
+  get dropSuggestion() {
+    return {
+      actual: this._totalDrops,
+      isGreaterThan: {
+        minor: 0,
+        average: 0,
+        major: 0,
```

Comment:
Is letting the buff drop so bad that it happening even once warrants a major suggestion? It might be worth spreading this out a bit (unless it really is that bad).

Reply:
well, I am more on the coder side than pure theorycrafter, but I play holy priest, and loosing your stacks is not only a pure loss of HPS, but also a mess to rebuild asap while also healing efficiently, and a huge mana loss. So I guess it is pretty bad overall ^^ I was hoping more expert holy players could give their point of view during the review still.

Reply:
![image](https://user-images.githubusercontent.com/9220340/119742687-ef363b00-be4d-11eb-8a0b-4530350bf671.png)
from mechanicalpriest.com